### ISSUE TITLE
Image pull from artifactory fails: Failed to obtain OAuth endpoint: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1 returned an unexpected response: Unexpected http code: 403

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#6738]
After successful login to artifcatory repo, image pull fails:
```
Nov  9 2017 14:58:46.410Z INFO  PullImage: reference: vic-docker-local.artifactory.eng.vmware.com/busybox:1, 127.0.0.1:2377, portlayer: "127.0.0.1:2377"
Nov  9 2017 14:58:46.410Z INFO  Using UUID (564d1c2f-2a64-983b-1ec9-4185f86f38c7) for imagestore name
Nov  9 2017 14:58:46.410Z DEBUG Running with portlayer
Nov  9 2017 14:58:46.410Z DEBUG [BEGIN]  [vic/lib/imagec.PingPortLayer:40] 127.0.0.1:2377
Nov  9 2017 14:58:46.411Z DEBUG [ END ]  [vic/lib/imagec.PingPortLayer:40] [385.047µs] 127.0.0.1:2377
Nov  9 2017 14:58:46.411Z DEBUG [BEGIN]  [vic/lib/imagec.LearnRegistryURL:75] vic-docker-local.artifactory.eng.vmware.com
Nov  9 2017 14:58:46.411Z DEBUG Trying https scheme for &imagec.Options{Reference:(*reference.taggedRef)(0xc420653e60), Registry:"vic-docker-local.artifactory.eng.vmware.com", Image:"busybox", Tag:"1", Destination:"/tmp", Host:"127.0.0.1:2377", Storename:"564d1c2f-2a64-983b-1ec9-4185f86f38c7", Username:"vic-deployer", Password:"vmware!123", Token:(*fetcher.Token)(nil), Timeout:3600000000000, Outstream:(*ioutils.WriteFlusher)(0xc42059fa40), InsecureSkipVerify:false, InsecureAllowHTTP:true, ImageManifestSchema1:(*imagec.Manifest)(nil), ImageManifestSchema2:(*schema2.DeserializedManifest)(nil), ManifestDigest:"", RegistryCAs:(*x509.CertPool)(0xc4207daf30), Standalone:false, ImageStore:""}
Nov  9 2017 14:58:46.411Z DEBUG URL: https://vic-docker-local.artifactory.eng.vmware.com/v2/
Nov  9 2017 14:58:46.550Z DEBUG [ END ]  [vic/lib/imagec.LearnRegistryURL:75] [138.852068ms] vic-docker-local.artifactory.eng.vmware.com
Nov  9 2017 14:58:46.550Z DEBUG [BEGIN]  [vic/lib/imagec.LearnAuthURL:100] vic-docker-local.artifactory.eng.vmware.com/busybox:1
Nov  9 2017 14:58:46.550Z DEBUG URL: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1
Nov  9 2017 14:58:46.587Z DEBUG [ END ]  [vic/lib/imagec.LearnAuthURL:100] [37.614669ms] vic-docker-local.artifactory.eng.vmware.com/busybox:1
Nov  9 2017 14:58:46.587Z INFO  https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1 returned an unexpected response: Unexpected http code: 403, URL: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1
Nov  9 2017 14:58:46.587Z DEBUG [ END ]  [vic/lib/apiservers/engine/backends.(*Image).PullImage:322] [226.636362ms] 
Nov  9 2017 14:58:46.587Z ERROR Handler for POST /v1.21/images/create returned error: Failed to obtain OAuth endpoint: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1 returned an unexpected response: Unexpected http code: 403, URL: https://vic-docker-local.artifactory.eng.vmware.com/v2/busybox/manifests/1
```

[21-2-Artifactory-VCH-0-7754-container-logs.zip](https://github.com/vmware/vic/files/1458276/21-2-Artifactory-VCH-0-7754-container-logs.zip)

**Comment by mdubya66** [comment]
Making high priority as this is required for a key customer

**Comment by sflxn** [comment]
This is the result of trying to obtain the manifest,

```
$ curl -k https://vic-docker-local.artifactory.eng.vmware.com/v2/vic-deployer/busybox/manifests/latest
{"errors":[{"code":"UNAUTHORIZED","message":"The client does not have permission for manifest","detail":{"manifest":"vic-deployer/busybox/latest/manifest.json"}}]}
```

**Comment by sflxn** [comment]
I found the problem with pulling on Artifactory.  When we first request the manifest, we authenticate.  We use an URL that consist of the registry URL, the repo, and manifest.  Based on the Docker's documentation, this seems correct and works with Docker Hub, Google, etc.

Artifactory doesn't reply with 401 (unauthorized).  Instead, it replies with 403, permission denied.  If we use just the registry URL on the initial query, we do get a 401 back, along with bearer realm info.  Without this, we cannot obtain the token.  This is the same URL we use for docker login.  The docker spec isn't clear about which URL is correct.  Using just the registry URL does not work with docker hub.

For now, I've added a workaround.  If the path with [registry url, manifest, repo] fails, we try again with just registry url.
